File: problems/1501-1600/1556_digit-grouping-with-dots/solution.py
from typing import List, Optional


class Solution:
    def groupDigits(self, n: int) -> str:
        # Reverse the digit string, cut it into runs of three, join with
        # '.', then reverse back — the chunk boundaries land exactly on
        # multiples of three counted from the units digit.
        digits = str(n)[::-1]
        groups = [digits[i : i + 3] for i in range(0, len(digits), 3)]
        return ".".join(groups)[::-1]

File: problems/1501-1600/1556_digit-grouping-with-dots/solution.rs
impl Solution {
    pub fn group_digits(n: i32) -> String {
        // Reverse the digit string, cut it into runs of three, join with
        // '.', then reverse back — the chunk boundaries land exactly on
        // multiples of three counted from the units digit.
        let digits: Vec<u8> = n.to_string().bytes().rev().collect();
        let groups: Vec<String> = digits
            .chunks(3)
            .map(|chunk| String::from_utf8(chunk.to_vec()).unwrap())
            .collect();
        groups.join(".").chars().rev().collect()
    }
}
